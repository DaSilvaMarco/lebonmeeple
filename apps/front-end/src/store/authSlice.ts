import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { type AuthState, initialState, type User } from './types';

const authSlice = createSlice({
  name: 'auth',
  initialState,
  reducers: {
    login: (state, action: PayloadAction<{ user: User; token: string }>) => {
      state.isLoading = false;
      state.user = action.payload.user;
      state.token = action.payload.token;
      state.isAuthenticated = true;

      localStorage.setItem('authToken', action.payload.token);
      localStorage.setItem('user', JSON.stringify(action.payload.user));
    },

    logout: (state) => {
      state.user = null;
      state.token = null;
      state.isAuthenticated = false;
      state.isLoading = false;

      localStorage.removeItem('authToken');
      localStorage.removeItem('user');
    },

    updateUser: (state, action) => {
      console.log('updateUser payload:', action.payload);
      state.user = action.payload;
    },
  },
});

export const { login, logout, updateUser } = authSlice.actions;

export const selectUser = (state: { auth: AuthState }) => state.auth.user;
export const selectToken = (state: { auth: AuthState }) => state.auth.token;
export const selectIsAuthenticated = (state: { auth: AuthState }) =>
  state.auth.isAuthenticated;
export const selectIsLoading = (state: { auth: AuthState }) =>
  state.auth.isLoading;

export default authSlice.reducer;
