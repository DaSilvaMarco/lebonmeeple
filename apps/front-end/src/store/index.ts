import { configureStore } from '@reduxjs/toolkit';
import authReducer from './authSlice';

// Configuration du store Redux
export const store = configureStore({
  reducer: {
    auth: authReducer,
  },
  // Middleware par défaut avec serialization check désactivé pour les fonctions
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        // Ignorer ces actions qui peuvent contenir des fonctions
        ignoredActions: ['persist/PERSIST', 'persist/REHYDRATE'],
      },
    }),
});

// Types pour TypeScript
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

// Hooks typés pour TypeScript
export { useDispatch, useSelector } from 'react-redux';
export type { TypedUseSelectorHook } from 'react-redux';

// Hooks personnalisés avec types
import {
  useDispatch as useReduxDispatch,
  useSelector as useReduxSelector,
  TypedUseSelectorHook,
} from 'react-redux';

export const useAppDispatch = () => useReduxDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useReduxSelector;
