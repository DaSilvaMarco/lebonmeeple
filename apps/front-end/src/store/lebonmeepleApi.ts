import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { getApiBaseUrl } from '@/utils/api-config';

export const lebonmeepleApi = createApi({
  reducerPath: 'lebonmeepleApi',
  baseQuery: fetchBaseQuery({ baseUrl: getApiBaseUrl() }),
  endpoints: () => ({}),
  tagTypes: ['Posts'],
  keepUnusedDataFor: 1,
  refetchOnMountOrArgChange: true,
});
