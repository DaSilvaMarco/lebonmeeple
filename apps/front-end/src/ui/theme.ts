import { type ThemeConfig, extendTheme } from '@chakra-ui/react';

// Configuration du thème (mode sombre/clair)
const config: ThemeConfig = {
  initialColorMode: 'light',
  useSystemColorMode: false,
};

// Couleurs personnalisées
const colors = {
  brand: {
    50: '#e3f2fd',
    100: '#bbdefb',
    200: '#90caf9',
    300: '#64b5f6',
    400: '#42a5f5',
    500: '#2196f3', // Couleur principale
    600: '#1e88e5',
    700: '#1976d2',
    800: '#1565c0',
    900: '#0d47a1',
  },
  // Couleurs pour Le Bon Meeple (thème jeu de société)
  meeple: {
    50: '#fff3e0',
    100: '#ffe0b3',
    200: '#ffcc80',
    300: '#ffb74d',
    400: '#ffa726',
    500: '#ff9800', // Orange principal
    600: '#fb8c00',
    700: '#f57c00',
    800: '#ef6c00',
    900: '#e65100',
  },
  game: {
    50: '#f3e5f5',
    100: '#e1bee7',
    200: '#ce93d8',
    300: '#ba68c8',
    400: '#ab47bc',
    500: '#9c27b0', // Violet pour le gaming
    600: '#8e24aa',
    700: '#7b1fa2',
    800: '#6a1b9a',
    900: '#4a148c',
  },
};

// Polices personnalisées
const fonts = {
  heading:
    '"Poppins", -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif',
  body: '"Inter", -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif',
};

// Styles des composants
const components = {
  // Button: {
  //   baseStyle: {
  //     fontWeight: 'semibold',
  //     borderRadius: 'lg',
  //   },
  //   variants: {
  //     solid: {
  //       bg: 'brand.500',
  //       color: 'white',
  //       _hover: {
  //         bg: 'brand.600',
  //         transform: 'translateY(-2px)',
  //         boxShadow: 'lg',
  //       },
  //       _active: {
  //         bg: 'brand.700',
  //         transform: 'translateY(0)',
  //       },
  //     },
  //     meeple: {
  //       bg: 'meeple.500',
  //       color: 'white',
  //       _hover: {
  //         bg: 'meeple.600',
  //         transform: 'translateY(-2px)',
  //         boxShadow: 'lg',
  //       },
  //       _active: {
  //         bg: 'meeple.700',
  //         transform: 'translateY(0)',
  //       },
  //     },
  //     game: {
  //       bg: 'game.500',
  //       color: 'white',
  //       _hover: {
  //         bg: 'game.600',
  //         transform: 'translateY(-2px)',
  //         boxShadow: 'lg',
  //       },
  //       _active: {
  //         bg: 'game.700',
  //         transform: 'translateY(0)',
  //       },
  //     },
  //   },
  //   defaultProps: {
  //     variant: 'solid',
  //   },
  // },
  Card: {
    baseStyle: {
      container: {
        borderRadius: 'xl',
        boxShadow: 'sm',
        _hover: {
          boxShadow: 'md',
          transform: 'translateY(-2px)',
        },
        transition: 'all 0.2s',
      },
    },
  },
  Input: {
    variants: {
      filled: {
        field: {
          bg: 'gray.50',
          borderRadius: 'lg',
          _hover: {
            bg: 'gray.100',
          },
          _focus: {
            bg: 'white',
            borderColor: 'brand.500',
            boxShadow: '0 0 0 1px var(--chakra-colors-brand-500)',
          },
        },
      },
    },
    defaultProps: {
      variant: 'filled',
    },
  },
  Heading: {
    baseStyle: {
      fontWeight: 'bold',
    },
    variants: {
      page: {
        fontSize: { base: '2xl', md: '3xl' },
        color: 'gray.800',
        marginBottom: 6,
      },
      section: {
        fontSize: { base: 'xl', md: '2xl' },
        color: 'gray.700',
        marginBottom: 4,
      },
    },
  },
};

// Styles globaux
const styles = {
  global: {
    body: {
      bg: 'gray.50',
      color: 'gray.800',
    },
    '*::placeholder': {
      color: 'gray.400',
    },
    '*, *::before, &::after': {
      borderColor: 'gray.200',
    },
  },
};

// Breakpoints personnalisés
const breakpoints = {
  base: '0em',
  sm: '30em',
  md: '48em',
  lg: '62em',
  xl: '80em',
  '2xl': '96em',
};

// Création du thème final
const theme = extendTheme({
  config,
  colors,
  fonts,
  components,
  styles,
  breakpoints,
  space: {
    px: '1px',
    0.5: '0.125rem',
    1: '0.25rem',
    1.5: '0.375rem',
    2: '0.5rem',
    2.5: '0.625rem',
    3: '0.75rem',
    3.5: '0.875rem',
    4: '1rem',
    5: '1.25rem',
    6: '1.5rem',
    7: '1.75rem',
    8: '2rem',
    9: '2.25rem',
    10: '2.5rem',
    12: '3rem',
    14: '3.5rem',
    16: '4rem',
    20: '5rem',
    24: '6rem',
    28: '7rem',
    32: '8rem',
    36: '9rem',
    40: '10rem',
    44: '11rem',
    48: '12rem',
    52: '13rem',
    56: '14rem',
    60: '15rem',
    64: '16rem',
    72: '18rem',
    80: '20rem',
    96: '24rem',
  },
});

export default theme;
