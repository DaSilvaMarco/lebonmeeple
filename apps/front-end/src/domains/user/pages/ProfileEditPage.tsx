'use client';

import React, { useRef, useState } from 'react';
import {
  useToast,
  Flex,
  Box,
  Center,
  Heading,
  VStack,
  FormControl,
  FormLabel,
  Input,
  FormErrorMessage,
  useColorModeValue,
  Text,
  Button as ChakraButton,
} from '@chakra-ui/react';
import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';
import { useRouter } from 'next/navigation';

import { toastSuccess, toastError } from '@frontend/domains/shared/toat/toast';
import NotConnected from '@frontend/domains/shared/warning/NotConnected';
import { useAppSelector, useAppDispatch } from '@frontend/store/hook';
import { patchUser } from '../api/patch-user';
import { updateUser } from '../slice';
import { UserProfileFormData, userProfileUpdateSchema } from '../type';
import { convertToBase64 } from '@frontend/utils/convertToBase64';
import Button from '@frontend/domains/shared/button/components/Button';

const ProfileEditPage = () => {
  const [selectedFileName, setSelectedFileName] = useState<string | null>(null);
  const fileInputRef = useRef<HTMLInputElement>(null);

  const textColorPrimary = useColorModeValue('neutral.800', 'white');

  const { user, isAuthenticated, token } = useAppSelector(
    (state) => state.user,
  );
  const toast = useToast();
  const dispatch = useAppDispatch();
  const router = useRouter();

  const {
    register,
    handleSubmit,
    setValue,
    formState: { errors, isValid, isDirty },
  } = useForm<UserProfileFormData>({
    resolver: zodResolver(userProfileUpdateSchema),
    defaultValues: {
      username: user?.username || '',
      email: user?.email || '',
      avatar: user?.avatar || '',
    },
    mode: 'onChange',
  });

  const handleFileUpload = async (e: React.ChangeEvent<HTMLInputElement>) => {
    if (e.target.files && e.target.files[0]) {
      const file = e.target.files[0];
      const allowedTypes = ['image/jpeg', 'image/png', 'image/jpg'];
      if (!allowedTypes.includes(file.type)) {
        setSelectedFileName(null);
        setValue('avatar', '', { shouldValidate: true, shouldDirty: true });
        toastError(
          toast,
          "Mauvais type d'avatar",
          'Seuls les fichiers JPG, JPEG ou PNG sont autorisés.',
        );
        return;
      }
      try {
        setSelectedFileName(file.name);
        const base64 = await convertToBase64(file);
        setValue('avatar', base64, { shouldValidate: true, shouldDirty: true });
      } catch (error) {
        console.error('Erreur lors du téléchargement du fichier:', error);
      }
    }
  };

  const onSubmit = async (data: UserProfileFormData) => {
    try {
      if (!token) {
        toastError(
          toast,
          "Erreur d'authentification",
          'Vous devez être connecté pour modifier votre profil.',
        );
        return;
      }

      const newUser = await patchUser(data, token);

      dispatch(updateUser(newUser));
      toastSuccess(
        toast,
        'Modification réussie !',
        'Votre profil a été mis à jour.',
      );

      setTimeout(() => {
        router.push('/');
      }, 1000);
    } catch (error) {
      const errorMessage =
        error instanceof Error
          ? error.message
          : "Une erreur inattendue s'est produite";
      toastError(toast, 'Erreur de connexion', errorMessage);
    }
  };

  if (!isAuthenticated || !user) {
    return <NotConnected />;
  }

  return (
    <Flex justifyContent="center">
      <Box w={{ base: '100%', md: '40%' }}>
        <Center>
          <Heading>Modifier le profil</Heading>
        </Center>
        <form onSubmit={handleSubmit(onSubmit)}>
          <VStack spacing="4">
            <FormControl isInvalid={!!errors.username}>
              <FormLabel htmlFor="username">Pseudo</FormLabel>
              <Input
                {...register('username')}
                id="username"
                type="text"
                variant="filled"
              />
              <FormErrorMessage>
                {typeof errors.username?.message === 'string'
                  ? errors.username.message
                  : ''}
              </FormErrorMessage>
            </FormControl>

            <FormControl isInvalid={!!errors.email}>
              <FormLabel htmlFor="email">Email</FormLabel>
              <Input
                {...register('email')}
                id="email"
                type="email"
                variant="filled"
              />
              <FormErrorMessage>
                {typeof errors.email?.message === 'string'
                  ? errors.email.message
                  : ''}
              </FormErrorMessage>
            </FormControl>

            <FormControl isInvalid={!!errors.avatar}>
              <FormLabel
                htmlFor="avatar"
                color={textColorPrimary}
                fontWeight="semibold"
                fontSize="sm"
              >
                Avatar (optionnel)
              </FormLabel>
              <Box display="flex" alignItems="center" gap={4}>
                <input
                  id="avatar"
                  type="file"
                  accept="image/*"
                  ref={fileInputRef}
                  onChange={handleFileUpload}
                  style={{ display: 'none' }}
                  data-testid="avatar-file-input"
                />
                <Button
                  type="button"
                  color="primary"
                  handleClick={() => fileInputRef.current?.click()}
                  dataTestId="avatar-upload-btn"
                >
                  Choisir un fichier
                </Button>
                <Text
                  fontSize="sm"
                  color={selectedFileName ? textColorPrimary : 'neutral.400'}
                >
                  {selectedFileName || 'Aucun fichier sélectionné'}
                </Text>
              </Box>
              <FormErrorMessage fontSize="sm" mt={2}>
                {typeof errors.avatar?.message === 'string'
                  ? errors.avatar.message
                  : ''}{' '}
              </FormErrorMessage>
            </FormControl>

            <ChakraButton
              disabled={!(isValid && isDirty)}
              type="submit"
              width="100%"
              colorScheme="brand"
            >
              Modifier le profil
            </ChakraButton>
          </VStack>
        </form>
      </Box>
    </Flex>
  );
};

export default ProfileEditPage;
