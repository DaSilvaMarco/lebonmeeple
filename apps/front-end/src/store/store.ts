import { configureStore } from '@reduxjs/toolkit';
import userSlice from '@/domains/user/slice';
import { lebonmeepleApi } from './lebonmeepleApi';

export const store = configureStore({
  reducer: {
    user: userSlice,
    [lebonmeepleApi.reducerPath]: lebonmeepleApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(lebonmeepleApi.middleware),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
