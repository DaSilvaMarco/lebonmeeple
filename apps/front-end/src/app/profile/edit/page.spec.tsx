import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { Provider } from 'react-redux';
import { ChakraProvider } from '@chakra-ui/react';
import { configureStore } from '@reduxjs/toolkit';
import ProfileEditPageApp from './page';
import { lebonmeepleApi } from '@frontend/store/lebonmeepleApi';
import theme from '@frontend/ui/theme';

// Mock du domaine user
vi.mock('@frontend/domains/user/pages/ProfileEditPage', () => ({
  default: vi.fn(() => (
    <div data-testid="profile-edit-page">Mocked ProfileEditPage</div>
  )),
}));

// Import du composant mocké pour les types
import ProfileEditPage from '@frontend/domains/user/pages/ProfileEditPage';

// Configuration du store de test
const createTestStore = () => {
  return configureStore({
    reducer: {
      [lebonmeepleApi.reducerPath]: lebonmeepleApi.reducer,
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(lebonmeepleApi.middleware),
  });
};

// Composant wrapper pour les tests
const TestWrapper = ({ children }: { children: React.ReactNode }) => {
  const store = createTestStore();

  return (
    <Provider store={store}>
      <ChakraProvider theme={theme}>{children}</ChakraProvider>
    </Provider>
  );
};

describe('Profile Edit Page App - Complete Coverage', () => {
  beforeEach(() => {
    // Reset tous les mocks avant chaque test
    vi.clearAllMocks();

    // Remettre le mock par défaut pour ProfileEditPage
    const MockedProfileEditPage = vi.mocked(ProfileEditPage);
    MockedProfileEditPage.mockImplementation(() => (
      <div data-testid="profile-edit-page">Mocked ProfileEditPage</div>
    ));
  });

  describe('Basic Functionality', () => {
    it('should render the page successfully', () => {
      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should render without any props', () => {
      const { container } = render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(container).toBeDefined();
      expect(container.firstChild).toBeTruthy();
    });

    it('should call ProfileEditPage component', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(MockedProfileEditPage).toHaveBeenCalledWith({}, {});
    });

    it('should call ProfileEditPage exactly once', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(MockedProfileEditPage).toHaveBeenCalledTimes(1);
    });
  });

  describe('Component Structure', () => {
    it('should return JSX element', () => {
      const result = ProfileEditPageApp();

      expect(result).toBeTruthy();
      expect(React.isValidElement(result)).toBe(true);
    });

    it('should be a functional component', () => {
      expect(typeof ProfileEditPageApp).toBe('function');
    });

    it('should be synchronous (not async)', () => {
      expect(ProfileEditPageApp.constructor.name).toBe('Function');
      expect(ProfileEditPageApp.constructor.name).not.toBe('AsyncFunction');
    });

    it('should not require any parameters', () => {
      // La fonction ne devrait pas lever d'erreur sans paramètres
      expect(() => ProfileEditPageApp()).not.toThrow();
    });

    it('should accept no props', () => {
      const result = ProfileEditPageApp();
      expect(React.isValidElement(result)).toBe(true);
    });
  });

  describe('Domain Integration', () => {
    it('should integrate correctly with user domain', () => {
      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should maintain clean architecture', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Vérifier que ProfileEditPage est appelé correctement
      expect(MockedProfileEditPage).toHaveBeenCalledWith({}, {});
      expect(MockedProfileEditPage).toHaveBeenCalledTimes(1);
    });

    it('should delegate all logic to domain layer', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Le composant application ne fait que déléguer
      expect(MockedProfileEditPage).toHaveBeenCalled();
    });
  });

  describe('Component Lifecycle', () => {
    it('should mount and unmount correctly', () => {
      const { unmount } = render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();

      unmount();

      expect(screen.queryByTestId('profile-edit-page')).not.toBeInTheDocument();
    });

    it('should re-render without issues', () => {
      const { rerender } = render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();

      rerender(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should handle multiple mounts', () => {
      const { unmount: unmount1 } = render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
      unmount1();

      const { unmount: unmount2 } = render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
      unmount2();
    });
  });

  describe('Code Quality', () => {
    it('should be a simple wrapper component', () => {
      const componentCode = ProfileEditPageApp.toString();

      // Le composant ne devrait pas contenir de logique complexe
      expect(componentCode).not.toMatch(/if\s*\(/);
      expect(componentCode).not.toMatch(/switch/);
      expect(componentCode).not.toMatch(/for\s*\(/);
      expect(componentCode).not.toMatch(/while\s*\(/);
    });

    it('should follow single responsibility principle', () => {
      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Le composant ne fait qu'une chose : rendre ProfileEditPage
      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should have minimal complexity', () => {
      const componentCode = ProfileEditPageApp.toString();

      // Vérifier que c'est un composant simple
      expect(componentCode.split('\n').length).toBeLessThan(10);
    });

    it('should not contain business logic', () => {
      const componentCode = ProfileEditPageApp.toString();

      // Pas de logique métier dans le composant application
      expect(componentCode).not.toMatch(/useState/);
      expect(componentCode).not.toMatch(/useEffect/);
      expect(componentCode).not.toMatch(/api/i);
      expect(componentCode).not.toMatch(/fetch/i);
      expect(componentCode).not.toMatch(/axios/i);
    });
  });

  describe('DDD Architecture Compliance', () => {
    it('should act as application layer component', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Le composant application délègue au domaine
      expect(MockedProfileEditPage).toHaveBeenCalled();
    });

    it('should delegate business logic to domain layer', () => {
      const componentCode = ProfileEditPageApp.toString();

      // Le composant ne devrait pas contenir de logique métier complexe
      expect(componentCode).not.toMatch(/business.*logic/i);
      expect(componentCode).not.toMatch(/validation/i);
      expect(componentCode).not.toMatch(/authentication/i);
      expect(componentCode).not.toMatch(/profile.*management/i);
      expect(componentCode).not.toMatch(/edit.*logic/i);
    });

    it('should maintain separation of concerns', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Vérification de la séparation des responsabilités
      expect(MockedProfileEditPage).toHaveBeenCalledWith({}, {});
    });

    it('should follow clean architecture principles', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Le composant devrait déléguer au domaine ProfileEditPage
      expect(MockedProfileEditPage).toHaveBeenCalled();

      // Vérifier qu'il n'y a pas de dépendances externes inappropriées
      const componentCode = ProfileEditPageApp.toString();
      expect(componentCode).not.toMatch(/http/i);
      expect(componentCode).not.toMatch(/database/i);
      expect(componentCode).not.toMatch(/repository/i);
    });
  });

  describe('Props Interface and Types', () => {
    it('should work without any props', () => {
      const result = ProfileEditPageApp();
      expect(React.isValidElement(result)).toBe(true);
    });

    it('should work with TypeScript strict mode', () => {
      // Test de compatibilité TypeScript
      const result: React.ReactElement = ProfileEditPageApp();
      expect(React.isValidElement(result)).toBe(true);
    });

    it('should have correct function signature', () => {
      expect(ProfileEditPageApp.length).toBe(0); // Aucun paramètre requis
    });

    it('should be compatible with React component interface', () => {
      // Vérifier que le composant peut être utilisé comme React component
      const TestComponent = () => <ProfileEditPageApp />;

      render(
        <TestWrapper>
          <TestComponent />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });
  });

  describe('Component Naming and Export', () => {
    it('should export the component as default', () => {
      expect(ProfileEditPageApp).toBeDefined();
      expect(typeof ProfileEditPageApp).toBe('function');
    });

    it('should have correct component name', () => {
      expect(ProfileEditPageApp.name).toBe('App');
    });

    it('should be importable as default export', () => {
      // Le composant devrait être exporté par défaut
      expect(ProfileEditPageApp).toBeTruthy();
    });
  });

  describe('Next.js Page Integration', () => {
    it('should work as Next.js page component', () => {
      // Les pages Next.js retournent du JSX
      const result = ProfileEditPageApp();
      expect(React.isValidElement(result)).toBe(true);
    });

    it('should be compatible with Next.js routing', () => {
      // Simuler l'utilisation par Next.js
      const result = ProfileEditPageApp();
      expect(React.isValidElement(result)).toBe(true);
    });

    it('should handle Next.js page lifecycle', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(MockedProfileEditPage).toHaveBeenCalled();
    });

    it('should work with Next.js static routes', () => {
      // Test pour route statique /profile/edit
      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });
  });

  describe('Performance', () => {
    it('should render efficiently', () => {
      const startTime = performance.now();

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      const endTime = performance.now();
      const renderTime = endTime - startTime;

      // Vérifier que le rendu prend moins de 500ms
      expect(renderTime).toBeLessThan(500);
    });

    it('should handle multiple concurrent renders', () => {
      // Rendu concurrent
      const results = [
        ProfileEditPageApp(),
        ProfileEditPageApp(),
        ProfileEditPageApp(),
      ];

      results.forEach((result) => {
        expect(React.isValidElement(result)).toBe(true);
      });
    });

    it('should not cause memory leaks', () => {
      const { unmount } = render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();

      // Unmount ne devrait pas causer d'erreurs
      expect(() => unmount()).not.toThrow();
    });

    it('should be lightweight', () => {
      const componentString = ProfileEditPageApp.toString();

      // Le composant devrait être très léger
      expect(componentString.length).toBeLessThan(1000);
    });
  });

  describe('Error Handling and Edge Cases', () => {
    it('should handle React context correctly', () => {
      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should work with different providers', () => {
      const CustomWrapper = ({ children }: { children: React.ReactNode }) => {
        const store = createTestStore();
        return (
          <Provider store={store}>
            <ChakraProvider theme={theme}>{children}</ChakraProvider>
          </Provider>
        );
      };

      render(
        <CustomWrapper>
          <ProfileEditPageApp />
        </CustomWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should handle provider changes gracefully', () => {
      const { rerender } = render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();

      // Re-render avec un nouveau wrapper
      rerender(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should maintain component stability', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      // Premier rendu
      const { rerender } = render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      const firstCallCount = MockedProfileEditPage.mock.calls.length;

      // Re-render
      rerender(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Le composant devrait être re-rendu
      expect(MockedProfileEditPage.mock.calls.length).toBeGreaterThan(
        firstCallCount,
      );
    });
  });

  describe('User Profile Edit Context', () => {
    it('should handle user profile edit requirements', () => {
      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should delegate profile editing logic to domain', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Toute la logique d'édition de profil est dans le domaine
      expect(MockedProfileEditPage).toHaveBeenCalled();
    });

    it('should not handle form validation logic directly', () => {
      const componentCode = ProfileEditPageApp.toString();

      // Pas de logique de validation dans le composant application
      expect(componentCode).not.toMatch(/validation/i);
      expect(componentCode).not.toMatch(/validate/i);
      expect(componentCode).not.toMatch(/error/i);
      expect(componentCode).not.toMatch(/schema/i);
    });

    it('should not handle form submission logic', () => {
      const componentCode = ProfileEditPageApp.toString();

      // Pas de logique de soumission dans le composant application
      expect(componentCode).not.toMatch(/submit/i);
      expect(componentCode).not.toMatch(/onSubmit/i);
      expect(componentCode).not.toMatch(/handleSubmit/i);
      expect(componentCode).not.toMatch(/form/i);
    });

    it('should work regardless of user edit state', () => {
      // Le composant application ne devrait pas dépendre de l'état d'édition
      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should delegate authentication checks to domain', () => {
      const componentCode = ProfileEditPageApp.toString();

      // Pas de vérifications d'authentification dans le composant application
      expect(componentCode).not.toMatch(/auth/i);
      expect(componentCode).not.toMatch(/login/i);
      expect(componentCode).not.toMatch(/token/i);
      expect(componentCode).not.toMatch(/session/i);
      expect(componentCode).not.toMatch(/user/i);
    });
  });

  describe('Integration with Testing Environment', () => {
    it('should work with React Testing Library', () => {
      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should work with Vitest mocking', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(MockedProfileEditPage).toHaveBeenCalled();
    });

    it('should be compatible with test environment', () => {
      // Test que le composant fonctionne dans l'environnement de test
      expect(() => {
        render(
          <TestWrapper>
            <ProfileEditPageApp />
          </TestWrapper>,
        );
      }).not.toThrow();
    });

    it('should support snapshot testing', () => {
      const { container } = render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('Edit Page Specific Features', () => {
    it('should render edit-specific UI', () => {
      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Vérifier que le composant d'édition est rendu
      expect(screen.getByTestId('profile-edit-page')).toBeInTheDocument();
    });

    it('should delegate edit operations to domain layer', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Toutes les opérations d'édition sont dans le domaine
      expect(MockedProfileEditPage).toHaveBeenCalled();
    });

    it('should not contain inline edit logic', () => {
      const componentCode = ProfileEditPageApp.toString();

      // Pas de logique d'édition en ligne
      expect(componentCode).not.toMatch(/edit/i);
      expect(componentCode).not.toMatch(/update/i);
      expect(componentCode).not.toMatch(/save/i);
      expect(componentCode).not.toMatch(/change/i);
    });

    it('should maintain edit context through domain', () => {
      const MockedProfileEditPage = vi.mocked(ProfileEditPage);

      render(
        <TestWrapper>
          <ProfileEditPageApp />
        </TestWrapper>,
      );

      // Le contexte d'édition est maintenu par le domaine
      expect(MockedProfileEditPage).toHaveBeenCalledWith({}, {});
    });
  });
});
