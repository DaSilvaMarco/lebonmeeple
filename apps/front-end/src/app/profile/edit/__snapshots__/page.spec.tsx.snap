// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Profile Edit Page App - Complete Coverage > Integration with Testing Environment > should support snapshot testing 1`] = `
<div
  data-testid="profile-edit-page"
>
  Mocked ProfileEditPage
</div>
`;
